### PR TITLE
implementedSearchBook

### PR DESCRIPTION
**Pull request by shumYevh**

### CODE REVIEW

File: src/main/java/org/example/bookweb/repository/BookSpecificationBuilder.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private final SpecificationProviderManager<Book> bookProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParameters searchParameters) {
+        Specification<Book> spec = Specification.where(null);
+        if (searchParameters.titles() != null && searchParameters.titles().length > 0) {
+            spec = spec.and(bookProviderManager.getSpecificationProvider("title")
```
Comment (Vasul-Pankiv) [inline src/main/java/org/example/bookweb/repository/BookSpecificationBuilder.java]: better to make spec_key a constant

File: src/main/java/org/example/bookweb/repository/book/AuthorSpecificationProvider.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    public static final String KEY = "author";
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    public Specification<Book> getSpecification(String[] params) {
+        return (root, query, criteriaBuilder)
+                -> root.get("author").in(Arrays.stream(params).toArray());
```
Comment (Vasul-Pankiv) [inline src/main/java/org/example/bookweb/repository/book/AuthorSpecificationProvider.java]:
```suggestion
                -> root.get(KEY).in(Arrays.stream(params).toArray());
```

File: src/main/java/org/example/bookweb/repository/book/IsbnSpecificationProvider.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    public static final String KEY = "isbn";
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    public Specification<Book> getSpecification(String[] params) {
+        return (root, query, criteriaBuilder)
+                -> root.get("isbn").in(Arrays.stream(params).toArray());
```
Comment (Vasul-Pankiv) [inline src/main/java/org/example/bookweb/repository/book/IsbnSpecificationProvider.java]: same

File: src/main/java/org/example/bookweb/repository/book/TitleSpecificationProvider.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    public static final String KEY = "title";
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    public Specification<Book> getSpecification(String[] params) {
+        return (root, query, criteriaBuilder)
+                -> root.get("title").in(Arrays.stream(params).toArray());
```
Comment (Vasul-Pankiv) [inline src/main/java/org/example/bookweb/repository/book/TitleSpecificationProvider.java]: same

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**